Automattic/wp-calypso
# WooExpress: Update eCommerce plans to show availability to upgrade from eCommerce trial

#### Proposed Changes

* Add potential ecommerce trial upgrade paths by adding the trial under paid ecommerce plans' `availableFor` methods.

**Before**

Note the lack of an upgrade path to ecommerce from the ecommerce trial

<img width="1451" alt="Screen Shot 2023-01-30 at 12 52 24 PM" src="https://user-images.githubusercontent.com/2124984/215555206-58c5e0b7-4596-47cc-b4a3-f04ab02862c2.png">

**After**

Upgrade is only available to ecommerce, no other plans

<img width="1476" alt="Screen Shot 2023-01-30 at 12 46 55 PM" src="https://user-images.githubusercontent.com/2124984/215554911-90cf5909-bf16-4ac7-ac70-4d95457061de.png">


#### Testing Instructions

* Switch to this PR, navigate to `/plans/[siteSlug]` on an ecommerce trial site
* You should see a button to upgrade only on the eCommerce plan (It incorrectly says "Upgrade to Yearly", that logic will need to be fixed in another PR)

#### Pre-merge Checklist



- NA Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- NA Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- NA For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?


Related to #72772
